lib/vintage_net_mobile/modems/quectel_BG96.ex

```diff
@@ -45,8 +45,8 @@ defmodule VintageNetMobile.Modems.QuectelBG96 do
   @impl true
   def add_raw_config(raw_config, config, opts) do
     ifname = raw_config.ifname
+    [%{apn: apn}] = config.service_providers
```

Comment:
Can we be more flexible and instead of requiring one and only one provider, can this take the first one? Later on, I think it would be useful for this to be a search through the list and pull out the APN that's appropriate for the modem and location (for example).